src/smolagents/models.py

```diff
@@ -1074,6 +1074,12 @@ def __call__(
         return self.postprocess_message(first_message, tools_to_call_from)
 
 
+class HfApiModel(InferenceClientModel):
+    """HfApiModel has been renamed to InferenceClientModel to more closely follow the name of the underlying Inference library."""
+
+    pass
```

Comment:
```suggestion
class HfApiModel(InferenceClientModel):
    def __new__(cls, *args, **kwargs):
        warnings.warn(
            "HfApiModel has been renamed to InferenceClientModel to more closely follow the name of the underlying Inference library.",
            DeprecationWarning,
        )
        return super().__new__(cls, *args, **kwargs)
```

I would rather set it as a warning otherwise users won't notice

(requires to `import warnings` in the import section)